tests/models/idefics3/test_image_processing_idefics3.py

```diff
@@ -0,0 +1,311 @@
+        self.num_channels = num_channels
+        self.num_images = num_images
+        self.image_size = image_size
+        self.min_resolution = min_resolution
+        self.max_resolution = max_resolution
+        self.do_resize = do_resize
+        self.size = size
+        self.do_normalize = do_normalize
+        self.image_mean = image_mean
+        self.image_std = image_std
+        self.do_rescale = do_rescale
+        self.rescale_factor = rescale_factor
+        self.do_convert_rgb = do_convert_rgb
+        self.do_pad = do_pad
+        self.do_image_splitting = do_image_splitting
+
+    def prepare_image_processor_dict(self):
+        return {
+            "do_convert_rgb": self.do_convert_rgb,
+            "do_resize": self.do_resize,
+            "size": self.size,
+            "do_rescale": self.do_rescale,
+            "rescale_factor": self.rescale_factor,
+            "do_normalize": self.do_normalize,
+            "image_mean": self.image_mean,
+            "image_std": self.image_std,
+            "do_pad": self.do_pad,
+            "do_image_splitting": self.do_image_splitting,
+        }
+
+    def get_expected_values(self, image_inputs, batched=False):
+        """
+        This function computes the expected height and width when providing images to BridgeTowerImageProcessor,
+        assuming do_resize is set to True with a scalar size and size_divisor.
+        """
+        if not batched:
+            shortest_edge = self.size["shortest_edge"]
+            longest_edge = self.size["longest_edge"]
+            image = image_inputs[0]
+            if isinstance(image, Image.Image):
+                w, h = image.size
+            elif isinstance(image, np.ndarray):
+                h, w = image.shape[0], image.shape[1]
+            else:
+                h, w = image.shape[1], image.shape[2]
+
+            aspect_ratio = w / h
+            if w > h and w >= longest_edge:
+                w = longest_edge
+                h = int(w / aspect_ratio)
+            elif h > w and h >= longest_edge:
+                h = longest_edge
+                w = int(h * aspect_ratio)
+            w = max(w, shortest_edge)
+            h = max(h, shortest_edge)
+            expected_height = h
+            expected_width = w
+        else:
+            expected_values = []
+            for images in image_inputs:
+                for image in images:
+                    expected_height, expected_width = self.get_expected_values([image])
+                    expected_values.append((expected_height, expected_width))
+            expected_height = max(expected_values, key=lambda item: item[0])[0]
+            expected_width = max(expected_values, key=lambda item: item[1])[1]
+
+        return expected_height, expected_width
+
+    def expected_output_image_shape(self, images):
+        height, width = self.get_expected_values(images, batched=True)
+        effective_nb_images = self.num_images * 5 if self.do_image_splitting else 1
```

Comment:
updated the tests!

Reply:
I still don't think this is correct - with the current image processor we're not guaranteed 5 patches if `do_image_splitting` is `True`

Reply:
No, we are not, but it is 5 with the rest of the parameters set arbitrarily throughout these tests (`min_resolution=30, max_resolution=40, longest_edge=20` -> 4 patches + 1 global-img). Would you recommend that I change the value in the tests where this function is called, to avoid confusion in the future? Otherwise, I can add a comment here explaining why it is five.